test/Generators/Microsoft.Gen.Logging/TestClasses/SensitiveRecordExtensions.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Compliance.Testing;
+using Microsoft.Extensions.Logging;
+
+namespace TestClasses
+{
+    [SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "Test code")]
+    [SuppressMessage("CodeQuality", "IDE0051:Remove unused private members", Justification = "Test code")]
+    [SuppressMessage("CodeQuality", "IDE0052:Remove unread private members", Justification = "Test code")]
+    [SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "Test code")]
+    [SuppressMessage("Major Code Smell", "S2376:Write-only properties should not be used", Justification = "Test code")]
+    internal static partial class SensitiveRecordExtensions
+    {
+        internal const string Sensitive = "SENSITIVE";
+
+        public record class BaseRecord
+        {
+            [PrivateData] public const string ConstFieldBase = Sensitive;
```

Comment:
What does it mean to apply this attribute to a const? Seems like we might need an analyzer to flag this as not desirable?

Reply:
I agree it doesn't make sense, but I wanted to list all possible combinations (even if they don't make sense) to ensure that `record`'s default `ToString()` implementation doesn't emit these values. I will file an issue for the analyzer shortly.

Reply:
Created #4669